index.js

```diff
@@ -223,6 +223,11 @@ function buildRouteCompress (fastify, params, routeOptions, decorateOnly) {
     if (payload == null) {
       return next()
     }
+    const responseEncoding = reply.getHeader('Content-Encoding')
+    if (responseEncoding && responseEncoding !== 'identity') {
+      // response is already compressed
+      return next()
+    }
     setVaryHeader(reply)
```

Comment:
Should the new block be after setting the setVaryHeader.

Reply:
Shouldn't fastify-static be responsible for setting the `Vary` header?

Reply:
I don't understand the answer. Anyway I'm landing - if it becomes a problem I guess we'll get another PR.